### ISSUE TITLE
FinalLocalVariable: false positive when variable is assigned inside and outside switch-block

### ISSUE THREAD

**Issue by MEZk** [checkstyle/checkstyle#3074]
Configuration:
```xml
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
          "-//Puppy Crawl//DTD Check Configuration 1.3//EN"
          "http://www.puppycrawl.com/dtds/configuration_1_3.dtd">

<module name="Checker">
    <property name="charset" value="UTF-8"/>

    <module name="TreeWalker">
<module name="FinalLocalVariable" />
    </module>
</module>
```

Input:
```java
public class Test {
private static void combine(final int[][] m, final int x, final int y, final int d, final int[] regionSizes) {
    final int a = m[y][x] & 1;
    int b;
    switch (d) {
    case 1:
      b = m[y - 1][x];
      break;
    case 2:
      b = m[y][x - 1];
      break;
    case 3:
      b = m[y][x + 1];
      break;
    default: 
      b = m[y + 1][x];
      break;
    }
    b &= 5;
  }
}
```

Output:
```
Starting audit...
[ERROR] /home/andreiselkin/Downloads/Test.java:5:9: Variable 'b' should be declared final. [FinalLocalVariable]
Audit done.
Checkstyle ends with 1 errors.

```

Variable b cannot be declared final, due to compile time error.
